Adding docs for new Influx 2.0 capabilities

mdegat01 opened the pull request:
Documentation additions to go along with [Influx 2.0 PR](https://github.com/home-assistant/core/pull/35392).  Also small adjustments to the 1.xx documentation to mention the fact that the `where` option in the sensor supports templates and add a new `database` option to the sensor platform to allow you to change the default db for all sensors.

## Proposed change
<!-- 
    Describe the big picture of your changes here to communicate to the
    maintainers why we should accept this pull request. If it fixes a bug
    or resolves a feature request, be sure to link to that issue in the 
    additional information section.
-->

This is a corresponding documentation update to along with the [Influx 2.0 PR](https://github.com/home-assistant/core/pull/35392).  See details on that PR for reasoning of the new feature addition.

## Type of change
<!--
    What types of changes does your PR introduce to our documentation/website?
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR.
-->

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information
<!--
    Details are important, and help maintainers processing your PR.
    Please be sure to fill out additional details, if applicable.
-->

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/35392
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist
<!--
    Put an `x` in the boxes that apply. You can also fill these out after
    creating the PR. If you're unsure about any of them, don't hesitate to ask.
    We're here to help! This is simply a reminder of what we are going to look
    for before merging your code.
-->

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

mdegat01 committed:
Adding docs for new Influx 2.0 capabilities

Corresponding doc PR for new Influx 2.0 capabilities added with [dev PR](https://github.com/home-assistant/core/pull/35392)

mdegat01 committed:
Fixed issues with YAML examples

Improper indentation snuck in examples, fixed it

klaasnicolaas commented on source/_integrations/influxdb.markdown:
```diff
@@ -37,41 +40,63 @@ You will still need to create a database named `home_assistant` via InfluxDB's c
 host:
   type: string
-  description: IP address of your database host, e.g., 192.168.1.10
+  description: **1.xx only** IP address of your database host, e.g., 192.168.1.10
```
If you start the description with a `*`, you need to quote from begin to end with `" "`

```suggestion
  description: "**1.xx only** IP address of your database host, e.g., 192.168.1.10"
```

mdegat01 replied:
Fixed. And thank you!  That error message was driving me crazy, I did not know what it was talking about with the line 4 column 16 bit.

mdegat01 replied:
@klaasnicolaas this issue should be resolved now but I can't seem to mark it as such, still showing up as a requested change blocking merge. Can you resolve this one? Just trying to get everything up for a merge now that the implementation is nearing completion.

mdegat01 committed:
Wrapped descriptions with stars in quotes

Wrapped descriptions starting with stars in quotes as requested for task

mdegat01 committed:
Fixed issues with required fields

Updated v2 only required fields to inclusive to meet validation requirements, wasn't aware they were type checked.  Moved invalid extra text into description.

mdegat01 committed:
Set api_v2 to default to false

Adjusted to correct error from netlify output

mdegat01 committed:
URL is not valid type

Adjusted to fix error in netlify output

mdegat01 committed:
Wrapped config with control characters in quotes

Trying to fix a 'mapping values are not allowed in this context' error from netlify.  Believe its because I have unquoted config descriptions with control characters

mdegat01 committed:
Unescaped quote was making config invalid

Needed to escape a quote inside a quoted description with control characters

mdegat01 committed:
Missed a type url in configuration

Fixing another Netlify reported issue

klaasnicolaas committed:
:pencil2: Tweak

frenck commented on source/_integrations/influxdb.markdown:
```diff
@@ -37,41 +39,64 @@ You will still need to create a database named `home_assistant` via InfluxDB's c
-  description: Path to use if your InfuxDB is running behind an reverse proxy.
+  description: "**1.xx only** Path to use if your InfuxDB is running behind an reverse proxy."
   required: false
+api_v2:
+  type: boolean
+  description: "**2.xx only** Set to `true` if connecting to a 2.xx installation, do not use otherwise."
+  required: inclusive
+  default: false
```
So what if v3 comes out? This feels confusing. I feel that there should be a single option which takes a version as a parameter, defaulting to v1.

mdegat01 replied:
The thread [here](https://github.com/home-assistant/core/pull/35392#pullrequestreview-421997905) has the reasoning behind this. I agree with the logic and at first I tried to implement it this way but discovered if I switch to a version field that takes a number there's a validation issue I'm not sure how to deal with.  Waiting on a second opinion in the other PR there to see how to best resolve this.

mdegat01 replied:
@frenck Issue should be resolved now. A solution to the validation issue was found so now both doc and runtime are updated to use an `api_version` config option instead of the v2 specific `api_v2`.

frenck commented on source/_integrations/influxdb.markdown:
```diff
@@ -37,41 +39,64 @@ You will still need to create a database named `home_assistant` via InfluxDB's c
+  description: "**2.xx only** Set to `true` if connecting to a 2.xx installation, do not use otherwise."
+  required: inclusive
+  default: false
+url:
+  type: string
+  description: "**2.xx only** URL of your 2.xx installation (e.g. http://localhost:9999)."
+  required: false
+  default: "https://us-west-2-1.aws.cloud2.influxdata.com"
```
Can't this be merged with the host/port configuration of v1?
Basically, even v1 API is an URL endpoint

mdegat01 replied:
Yep you're right, modified the two PRs accordingly.  `url` option is gone, building the URL from the other config options.

mdegat01 committed:
Updating config options based on PR feedback

PR feedback suggested to drop the `url` option and instead use the existing `ssl`, `host`, `port` and `path` options to build the URL for V2.  Adjusted doc to reflect this new configuration.

mdegat01 committed:
Replaced api_v2 with api_version config option

Validation issue with implementation has been resolved therefore updating config to use the preferrable `api_version` config option instead of an option specifically for `api_v2`. Doc is now up to date with latest feature commit.

mdegat01 commented:
Implementation PR has now been merged. Please let me know if any other changes are needed here so the doc can go in as well.

frenck commented on source/_integrations/influxdb.markdown:
```diff
@@ -35,43 +37,60 @@ influxdb:
 host:
   type: string
-  description: IP address of your database host, e.g., 192.168.1.10
+  description: IP address or domain of your database host, e.g., 192.168.1.10. **2.xx - Defaults to 'us-west-2-1.aws.cloud2.influxdata.com' for 2.xx, not 'localhost'.**
```
This does not need emphasis, just write it in the normal text would do.

The same applies to all other occasions as well.

mdegat01 replied:
Ok will remove the emphasis.

frenck commented on source/_integrations/influxdb.markdown:
```diff
@@ -197,79 +249,193 @@ sensor:
       default: value
 {% endconfiguration %}
 
+### Configuration for 2.xx Installations
```
Why is this repeated with all options again?

I'm confused... a newer user would be as well I guess.

mdegat01 replied:
My thinking was that it was different enough for 2.xx with the entirely different query language that it would be easier to just separate 1.xx and 2.xx configurations and examples. But I can merge and just put in `2.xx only` for things like I did other places.

mdegat01 replied:
Ok merged the config sections, let me know what you think

mdegat01 committed:
Removed emphasis and merged v1 and v2 config

Updated documentation per suggested feedback

frenck committed:
:pencil2: Tweaks

frenck reviewed (approved):
Thanks, @mdegat01 👍
